tests/test_requests.py

```diff
@@ -789,6 +789,15 @@ def test_status_raising(self, httpbin):
         r = requests.get(httpbin('status', '500'))
         assert not r.ok
 
+    def test_status_raising_hides_password(self, httpbin):
+        host = urlparse(httpbin()).netloc
+
+        r = requests.get('http://user:pass@' + host + '/status/404')
+        with pytest.raises(requests.exceptions.HTTPError) as e:
+            r.raise_for_status()
+
+        assert 'pass' not in str(e)
```

Comment:
💡 I'd rather assert e is equals with the string with pass obfuscated, and also add a test that there doesn't obfuscate things that wasn't suppose to, but I'm happy to approve it.

Reply:
I've made the test more specific now. How would you implement a test that checks if other things are not obfuscated?

Reply:
You can create a test like this:
```
    def test_status_raising_doesnt_hides_other_stuff(self, httpbin):
        host = urlparse(httpbin()).netloc

        r = requests.get('http://shouldntchange@' + host + '/status/404')
        with pytest.raises(requests.exceptions.HTTPError) as e:
            r.raise_for_status()

        host_without_port = host.split(':')[0]

        assert str(e.value) == '404 Client Error: NOT FOUND for url: http://shouldntchange@' + host_without_port + '/status/404'
```

This is just a validation that the match doesn't put wrong information if the password is not supplied